src/sql/parts/notebook/models/clientSession.ts

```diff
@@ -302,7 +302,7 @@ export class ClientSession implements IClientSession {
 	 */
 	public async shutdown(): Promise<void> {
 		// Always try to shut down session
-		if (this._session && this._session.id) {
+		if (this._session && this._session.id && this.notebookManager.sessionManager.isReady) {
```

Comment:
So I looked at the JupterSessionManager and it no-ops if session isn't ready anyhow. This one therefore shouldn't be doing anything / shouldn't be a cause of errors.

Reply:
Thanks for checking this.